lib/fault_handler/fault_handler.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <arch/cpu.h>
+#include <logging/log_ctrl.h>
+#include <logging/log.h>
+#include <fatal.h>
+
+LOG_MODULE_REGISTER(fault_handler, CONFIG_FAULT_HANDLER_LOG_LEVEL);
+
+__weak void system_reset(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	NVIC_SystemReset();
+}
+
+void k_sys_fatal_error_handler(unsigned int reason,
+			       const z_arch_esf_t *esf)
+{
+	ARG_UNUSED(esf);
+
+	LOG_PANIC();
+
+	LOG_ERR("Resetting system");
+	system_reset(reason);
```

Comment:
Could you call, here, sys_arch_reboot(int), which is already weak, and does pretty much the same?

Reply:
I left `system_reset `as weak but inside it `sys_arch_reboot(int)` is used now

Reply:
I think you should just call sys_arch_reboot(..) directly here, unless you have identified a reason not to.